### PR TITLE
Updated to work with Python >3.6 + OTP authentication

### PR DESCRIPTION
**Pull request by ashwinpra**
Made the following changes:
- Removed unnecessary libraries from `requirements.txt`
- Bumped up other libraries
- Replaced deprecated `_attr_value_as_string` method of bs4
- Replaced `is` with `==` (It throws `SyntaxWarning` with some types of literals from Python >=3.8)
- Updated README

### CODE REVIEW

**Comment by ashwinpra** [comment]
ERP recently introduced OTP authentication for login, so I have made required changes to the code for it to work
image

**Review by proffapt — APPROVED** [review]
LGTM. Tested locally, works.

**Review by proffapt — APPROVED** [review]

**Comment by ashwinpra** [comment]
- After the last update, the [iitkgp-erp-login](https://github.com/proffapt/iitkgp-erp-login-pypi/) package was made, so I incorporated that into the script
- This package already asks the user for their roll number, so I removed the corresponding command-line argument (`--user`)
- Updated dates for Spring 2023-24 semester
- Added 3 courses to `subjects.json`:    
    - AI61003: LINEAR ALGEBRA FOR AI AND ML
    - CS60029: RANDOMIZED ALGORITHM DESIGN
    - CS60075: NATURAL LANGUAGE PROCESSING

**Review by proffapt — CHANGES_REQUESTED** [review]
- I don't know how `gyft-serve` works, but integrating my package here will require a bit of extra work to be done on `gyft-serve` to get it working.
- Also date updates have a separate PR #87

File: requirements.txt [hunk 1]
```diff
@@ -1,24 +1,8 @@
...
-rsa==4.0
-simplejson==3.17.0
-six==1.13.0
-uritemplate==0.6
-urllib3==1.24.2
+beautifulsoup4==4.12.2
+google_api_python_client==2.90.0
+httplib2==0.22.0
+icalendar==5.0.7
+iitkgp_erp_login==2.1.3
```
Comment (proffapt) [inline requirements.txt:5]:
```suggestion
iitkgp_erp_login==2.2.0
```

File: gyft.py [hunk 2]
```diff
@@ -5,21 +5,12 @@
...
-	args.user = input("Enter you Roll Number: ")
-erp_password = getpass.getpass("Enter your ERP password: ")
-
-#### Parsing ends
+import iitkgp_erp_login.erp as erp
 
 ERP_HOMEPAGE_URL = 'https://erp.iitkgp.ac.in/IIT_ERP3/'
 ERP_LOGIN_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/auth.htm'
 ERP_SECRET_QUESTION_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getSecurityQues.htm'
+ERP_OTP_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getEmilOTP.htm' # blame ERP for the typo
```
Comment (proffapt) [inline gyft.py:13]:
```suggestion
```
The package already includes these, just check once whether they are used else where or not ( I don't see them )

**Review by proffapt — CHANGES_REQUESTED** [review]

File: gyft.py (hunk 2 above)
Comment (ashwinpra) [reply gyft.py:13]: You're right, they're not required anymore, I hadn't noticed it.

File: subjects.json [hunk 3]
```diff
@@ -679,10 +682,10 @@
     "IM21201": "OPERATIONS RESEARCH-I",
     "IM21203": "ENGINEERING ECONOMY",
     "IM22001": "INFORMATION SYSTEMS LAB",
-    "IM29202": "WORK SYSTEM DESIGN LABORATORY",
-    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
     "IM29006": "OPERATIONS RESEARCH LAB.",
     "IM29008": "WORK SYSTEM DESIGN LAB.",
+    "IM29202": "WORK SYSTEM DESIGN LABORATORY",
+    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
```
Comment (proffapt) [inline subjects.json:688]:
```suggestion
    "IM29202": "WORK SYSTEM DESIGN LABORATORY",
    "IM29204": "OPERATIONS RESEARCH-II LABORATORY",
```
This change distorts the sorted order of courses, please correct it.

**Review by proffapt — CHANGES_REQUESTED** [review]
After the requested changes, CLI will be good to go.

File: subjects.json (hunk 3 above)
Comment (proffapt) [reply subjects.json:688]: Not needed

**Comment by proffapt** [comment]
@chirag-ghosh This PR is good to be merged, works like a charm. With my package all the possible cases of login are handled (tested robustly). It was only aimed at CLI applications but I will later add support for web apps. Before that I will add a temporary fix on gyft-serve.

**Review by chirag-ghosh — APPROVED** [review]

**Pull request merged**